### DISCUSSION TITLE
Can no longer override temporary environment variables set in test runs

### DISCUSSION THREAD

**Discussion by brdly** [vitest-dev/vitest#4247] [Q&A] [answered]
Hey, we're currently in the process of upgrading from version 0.33.0 to 0.34.6 and a couple of our tests have broken in unexpected ways. Before raising a bug report we wanted to query if this behaviour is as expected and if so how we should go about fixing it.

We have this code pre tests:
```
const VITE_ENVIRONMENT = null;
const VITE_VERSION = null;

beforeEach(() => {
  ...
  import.meta.env = { ...VITE_ENVIRONMENT, ...VITE_VERSION };
});
```

In our first test we make a call to an external function that uses the above two environment variables and sets values that returns like below:
```
test('calls the init function with the correct parameters when env and version are not set', async () => {
  externalFunction();

  expect(spy).toHaveBeenCalledWith({
    ...
    env: 'localhost',
    version: 'localhost',
  });
});
```

This test passes fine when run first. The next test then changes the two environment variables and runs as below:
```
test('calls the init function with the correct parameters when env and version are set', async () => {
  import.meta.env.VITE_ENVIRONMENT = 'production';
  import.meta.env.VITE_VERSION = '1234';

  externalFunction();

  expect(spy).toHaveBeenCalledWith({
    ...
    env: 'production',
    version: '1234',
  });
});
```

This test fails as it uses the environment variables as set in the beforeEach and ignores our changes to the environment variables within the test. Is this as intended?

**Comment by sheremet-va** [comment, marked as answer]
Yes, this looks like expected behaviour. `import.meta.env` is scoped to the current module (`import.meta` should signal that), so when you override it, it only affects the current module, and I assume the `externalFunction` is defined in another file. To fix this, you can either use Node.js `process.env`:

```ts
process.env = { ...VITE_ENVIRONMENT, ...VITE_VERSION }
```

Or `Object.assign`:

```ts
Object.assign(import.meta.env, {  ...VITE_ENVIRONMENT, ...VITE_VERSION })
```

**Reply by brdly** [reply]
Thanks for the quick response, `process.env` works for us well here. :)
